procedures/automation-android-setup.md

```diff
@@ -0,0 +1,200 @@
+    2. Quite a few changes in the others: scopes, repositories, docker image
+    3. Once the rest of the infrastructure is set up, you can lean on "staging builds" to be confident that everything is configured correctly
+5. Create staging hook (e.g.: [`fenix-nightly-staging`](https://taskcluster-web.netlify.com/hooks/project-mobile/fenix-nightly-staging))
+    1. Go to [Hooks](https://taskcluster-web.netlify.com/hooks)
+    2. Click "Create Hook" at the bottom-right of the page
+    3. `HookGroupId` is "project-mobile"
+    4. `HookId` is `$product-nightly-staging` (e.g.: `fenix-nightly-staging`)
+    5. `Name` is "$product Nightly Builds (staging)"
+    6. Add a `Description` if it can be more descriptive than the `Name`
+    7. `Owner` is different for each product, ask the product's team which email address they'd like to receive build failure notifications from
+    8. Use the template from [`fenix-nightly`](https://taskcluster-web.netlify.com/hooks/project-mobile/fenix-nightly) with some modifications:
+        1. Use the correct `payload.image`
+        2. Update the `payload.command` to clone and run the correct repository
+        3. Update the `metadata`
+        4. `scopes` should be `assume:hook-id:project-mobile/$hookId` (using the `HookId` you defined above)
+6. Do the same thing, but for non-staging nightlies (e.g.: [`fenix-nightly`](https://taskcluster-web.netlify.com/hooks/project-mobile/fenix-nightly)):
+    1. `HookId` should be `$product-nightly`
+    2. `Name` should not end with "(staging)"
+    3. Copy the task template from the staging hook, with some modifications
+        1. Remove the `--staging` parameter from `payload.command`
+        2. Update `metadata` to remove anything mentioning "staging"
+        3. Update the scope to assume from the non-staging hook's role
+7. Create the role for the staging hook (e.g.: [`fenix-nightly-staging`](https://taskcluster-web.netlify.com/auth/roles/hook-id%3Aproject-mobile%2Ffenix-nightly-staging))
+    1. Go to [Roles](https://taskcluster-web.netlify.com/auth/roles)
+    2. `RoleId` should be idential to your staging hook's `HookId`, but prefixed with `hook-id:` (e.g.: `hook-id:project-mobile/fenix-nightly-staging`)
+    3. `Description` should be something like "Hook for building, signing and uploading (staging) Nightly versions of $product"
+    4. Add all the scopes you'll need to fulfill the requirements in you `.taskcluster.yml` file (when `is_mozilla_mobile_repo` is `false`)
+8. Do the same thing, but for non-staging nightlies (e.g.: [`fenix-nightly`](https://taskcluster-web.netlify.com/auth/roles/hook-id%3Aproject-mobile%2Ffenix-nightly))
+    1. Remove mentions of "staging" from each field
+    2. Make sure that all scopes are provided to fulfill requirements in `.taskcluster.yml` (when `is_mozilla_mobile_repo` is `true`)
+9. Update `scriptworker`, `pushapk` and `mozapkpublisher`
```

Comment:
I'm wondering if we should add a quick note like: _Should you need publishing an Android product to the Google Play store, you'l also need to ..._ Mainly I'm thinking at `android-components` or `application-services` which don't have this step but instead they require `beetmover` bits. However, I could be wrong as this docs overall is aimed for the former.

Reply:
Call me selfish, but I'd like to wait until we have our documentation situation sorted out. Once that's accomplished, we can restructure these docs to branch and be more flexible.

I'm guessing that, long-term, we'll have these split into sections like:
* Core automation
* PR checks
* Signing
* Deployment to Google Play
* Deployment to Amazon Store
* Deployment to Maven
* etc

What do you think?

Reply:
This is great, indeed! I fully agree 👍